zenoh/tests/interceptors.rs

```diff
@@ -0,0 +1,216 @@
+use std::sync::{Arc, Mutex};
+
+struct IntervalCounter {
+    first_tick: bool,
+    last_time: std::time::Instant,
+    count: u32,
+    total_time: std::time::Duration,
+}
+
+impl IntervalCounter {
+    fn new() -> IntervalCounter {
+        IntervalCounter {
+            first_tick: true,
+            last_time: std::time::Instant::now(),
+            count: 0,
+            total_time: std::time::Duration::from_secs(0),
+        }
+    }
+
+    fn tick(&mut self) {
+        let curr_time = std::time::Instant::now();
+        if self.first_tick {
+            self.first_tick = false;
+        } else {
+            self.total_time += curr_time - self.last_time;
+            self.count += 1;
```

Comment:
Is `IntervalCounter` be used in the code beside the tests?
`IntervalCounter` does not seem to provide a method to reset it and increment on `total_time` and `count` would lead to some overflow. As a side node, [wrapping_add](https://doc.rust-lang.org/std/primitive.usize.html#method.wrapping_add) could be used to deal with overflows.

Reply:
I have no any intention to use it outside of the tests, this class was made only to improve the readability of tests and make it easier to write new ones.
Thanks for the wrapping_add, I'll know about it.